tools/ansible/tasks/wsu/main.yaml

```diff
@@ -20,25 +20,160 @@
         msg: "Cluster not patched for hybrid overlay"
       when: "hybrid_query.stdout == ''"
 
+    # Expected value of kubernetes_version in the form of 'v1.17.2'
+    - name: Get kubernetes version
+      shell: "oc version -o json | jq -r '.serverVersion.gitVersion'"
+      register: kubernetes_version
+      failed_when: kubernetes_version.stdout == ""
+
+    # kubernetes_version v1.17.2 should correspond to 1.17 for getting changelog url
+    - name: Get kubernetes version without patch version
+      shell: "version={{ kubernetes_version.stdout }} ; echo ${version%.*} | tr -d v"
+      register: kubernetes_version_without_patch
+      failed_when: kubernetes_version_without_patch.stdout == ""
+
+    # Changelog url to lookup SHA for kube node binaries
+    # Example: https://github.com/kubernetes/kubernetes/blob/release-1.17/CHANGELOG/CHANGELOG-1.17.md
+    - name: Get kubernetes changelog url
+      shell: |
+        echo "https://github.com/kubernetes/kubernetes/blob/release-"\
+        "{{ kubernetes_version_without_patch.stdout }}/CHANGELOG/CHANGELOG-"\
+        "{{ kubernetes_version_without_patch.stdout }}.md"
+      register: changelog
+
+    - name: Get kube node binary SHA
+      shell: |
+        curl -s "{{ changelog.stdout }}" | \
+        grep -A 1 "{{ kubernetes_version.stdout }}" | \
+        grep -A 1 node-windows | tail -n 1 | sed -e 's/.*<code>\(.*\)<\/code>.*/\1/'
```

Comment:
What does this mean?Please add a comment here?

Reply:
I have added a comment here to give an overview of what we are trying to get